src/mpid/ch4/netmod/ofi/ofi_init.c

```diff
@@ -543,7 +544,9 @@ int MPIDI_OFI_mpi_init_hook(int rank, int size, int appnum, int *tag_bits, MPIR_
     /* ------------------------------------------------------------------------ */
 
     int num_vnis = 1;
-    if (MPIR_CVAR_CH4_OFI_MAX_VNIS > 1) {
+    if (MPIR_CVAR_CH4_OFI_MAX_VNIS == 0 || MPIR_CVAR_CH4_OFI_MAX_VNIS > MPIDI_global.n_vcis) {
+        num_vnis = MPIDI_global.n_vcis;
```

Comment:
Is `MPIR_CVAR_CH4_NUM_VCIS ` the only way to set `MPIDI_global.n_vcis`? (trying to see if the commit title represents all the cases)

Reply:
I believe that is the case.